### ISSUE TITLE
Question: What are the risks in using H.265?

### ISSUE THREAD

**Issue by hisaac** [donmelton/video_transcoding#229]
I see video_transcoding supports H.265 conversion, but that it's labeled as experimental. Could you explain the risks in using H.265? Is it too early to start using it?

**Comment by khaosx** [comment]
I've had some good luck with H.265, but in my opinion, it's not ready for prime time yet. While there aren't any risks, I can tell you some of the downsides from my experience.

* Time - It takes **for-freaking-ever** to transcode. On a 6 core Mac Pro, I can't transcode 30fps (generally considered real-time).
* Playback - It's a hodge podge. If your player supports it, you're golden. Plex is almost assuredly going to transcode it unless you've got a Shield.
* Tool chain - Not nearly as mature. I'll leave it to @donmelton to talk about the technical details if he chooses, but I had to completely retool my workflow to even start the testing. 

tl;dr? - Stay away for at least another year, unless you're starting from scratch on your transcoding journey, you've verified that ALL your targets playback H.265, and you have a beast of a transcoding box. Your mileage may vary, but this is the direction I'm taking for now...

**Comment by hisaac** [comment]
Thanks @khaosx! This is very helpful info. I think I'll take your advice and hold off for now.

**Comment by donmelton** [comment]
@hisaac This is a great question and @khaosx gave a great answer. I agree with all of that. Thanks!

The HEVC (or H.265) format is very possibly the future of video transcoding. Although there's also the possibility that [AV1](https://en.wikipedia.org/wiki/AV1) will wind up replacing H.264. We just don't know yet.

But while HEVC may be the future, IMO it's definitely not the present. :) At least for home transcoders like us.

While there are now some commercial software and hardware HEVC codecs, using HandBrake our choice is limited to x265. I think the x265 development team has made tremendous progress with performance and quality. But there are still issues.

As @khaosx wrote, using x265 is slow. More than twice as slow as using x264, the last time I checked. And I haven't been able to develop a `--quick` option for x265 yet which speeds things up and still maintains quality.

And as to quality, there are still a few problems with x265 even when using its default settings. I know of at least one very grainy scene in a movie where x265 produces significant blockiness and artifacts.

Plus, my special, or default, ratecontrol system in `transcode-video` is not tuned for x265. It works, but I'm sure I could improve it if I had the time and high-powered hardware to iterate on it.

And @khaosx is completely spot on with his concerns about playback. Only the latest, high-end devices can handle HEVC.

If you do decide to experiment with HEVC transcoding, my advice is to use my ABR ratecontrol system (via the `--abr` option) and the 10-bit version of the x265 codec (via `--handbrake-option encoder=x265_10bit`).

**Comment by hisaac** [comment]
Sounds good @donmelton, thank you for the technical breakdown and extra advice.
